crates/graphql-composition/src/subgraphs/directives/directive_definition.rs

```diff
@@ -0,0 +1,105 @@
+use super::*;
+use bitflags::bitflags;
+use std::fmt::Display;
+
+pub(crate) struct DirectiveDefinition {
+    pub(crate) subgraph_id: SubgraphId,
+    pub(crate) name: StringId,
+    pub(crate) locations: DirectiveLocations,
+    pub(crate) arguments: Vec<DirectiveDefinitionArgument>,
```

Comment:
It's a list of `InputValueDefinition` like field arguments, no need for extra type I think. but don't remember how you represent the rest in composition

Reply:
Hm true, it's even in the spec. Unlike graphql-federated-graph, we don't have a unified `InputValueDefinition` in graphql_composition::Subgraphs, they're scattered in different HashMaps and such, so I'll have to think about how we can deduplicate.

Reply:
But for now I'll rename it to InputValueDefinition at least.

Reply:
Also have to validate that it matches. We should also match the directives applied to the input value definitions.

Reply:
Done, and pushed.